docs/keyed_services.md

```diff
@@ -0,0 +1,184 @@
+### Keyed Services
+
+Most features should be scoped to a Profile/BrowserContext and be
+BrowserContextKeyedServices per
+[profile-architecture](https://www.chromium.org/developers/design-documents/profile-architecture/).
+What this means is that all associated preferences and other locally stored
+state will be per-profile. A service may sometimes be shared between the regular
+and OTR profile (check with sec-team) and it may also be ununavailable for
+certain profile types.
```

Comment:
having the same / different / null service for normal vs incognito profile is such a common source of confusion and errors that i would emphasize on that. Probably a separate section with it's own heading saying something like "think twice about how your feature / service should behave in incognito / tor / etc and ask if not sure"